### PR TITLE
fix(cli-repl): prompt for password when redirecting output MONGOSH-1136

### PR DESCRIPTION
**Pull request by mabaasit**

### CODE REVIEW

**Review by addaleax — APPROVED** [review]
LGTM, feel free to add a test but this is probably just fine as it is as well

File: packages/cli-repl/src/run.spec.ts [hunk 1]
```diff
@@ -97,4 +97,52 @@ describe('CLI entry point', function () {
...
+      const [code] = await once(proc, 'exit');
+      expect(code).to.equal(1);
+      expect(promptedForPassword).to.be.true;
+    });
+
+    it('requests password when user is redirecting output', async function () {
+      const args = [
+        ...pathToRun,
+        'mongodb://amy@localhost:27017',
+        '> /dev/null',
```
Comment (addaleax) [inline packages/cli-repl/src/run.spec.ts:127]: This doesn't work on Windows, does it? I think you'd need to end up doing something like `stdio: ['pipe', 'ignore', pipe']` or maybe actually opening a file and passing that fd as the stdout entry?

**Review by addaleax — APPROVED** [review]

File: packages/cli-repl/src/run.spec.ts (hunk 1 above)
Comment (mabaasit) [reply packages/cli-repl/src/run.spec.ts:127]: yep, it does not! will check it!

Comment (mabaasit) [reply packages/cli-repl/src/run.spec.ts]: cleaned up in [83511f0](https://github.com/mongodb-js/mongosh/pull/2116/commits/83511f0fc5e6faf0a8a76c8d2c86f00e5e271b7b)

**Review by addaleax — APPROVED** [review]

**Pull request merged**